warehouse/accounts/views.py

```diff
@@ -282,6 +283,9 @@ def two_factor_and_totp_validate(request, _form_class=TOTPAuthenticationForm):
                 .hexdigest()
                 .lower(),
             )
+
+            send_recovery_code_reminder_email(request, request.user)
```

Comment:
I don't see how this is conditional on _not_ having recovery codes... seems like it'd fire regardless?